### DISCUSSION TITLE
Dynamic creation of OpenAPI specifications based on user type

### DISCUSSION THREAD

**Discussion by evgenoir** [apiflask/apiflask#437] [Q&A]
Hi there,
Suppose I create a custom protected view for docs. Is it possible to hide some view function from API documentations (and OpenAPI spec) for some users? For example, if the administrator logs into the application, he sees the entire OpenAPI specification, but if the user logs in, he only sees a partial OpenAPI specification where some views are hidden. Is there a way to implement it in ApiFlask?

**Comment by davidjb99** [comment]
This would be a great feature!
